### ISSUE TITLE
Implement a way to define average windows waveform PVs

### ISSUE THREAD

**Issue by jesusvasquez333** [cristinasewell/ATCM_PyDM_Screens#62]
Under the `LLRF HLS` screen, there are 6 windows waveforms PVs:
- Average Window 0 to 2
- Complex window 0 to 2

These windows are waveform PVs. We need to add a tool here to let the user generate these PVs from some parameters. For example, for a rectangular window, the user need to define the `start` and `stop` time, and from these 2 parameters the waveform PV data should be automatically generated and  `caput`ed to the corresponding PV.

We should also support some other standard windows functions, like some of [theses](https://en.wikipedia.org/wiki/Window_function#A_list_of_window_functions). We should decide which ones to support, and which parameters to use to generated them.

**Comment by jesusvasquez333** [comment]
Let me give a more elaborate example:
I image somewhere in the `LLRF HLS` screen a button labeled `define window`, which will open a new windows with some widgets that will allow the user to define the window function and finally write the resulting windows to the waveform PV. 
In this windows, we will have:
- A menu to select the type of window (Rectangular, Sine, Hann, ...)
- text boxes to introduce the windows parameters (they will depend on the type of windows selected above),
- a preview plot of the window, updated when the type/parameters are changed (this plot a local copy of the windows data, not the real waveform PV),
- a button labeled `write` to write the data to the waveform PV.
- Maybe also the real waveform PV? Maybe on the same plot use to preview the new window function.

For example, for a rectangular windows, the user will introduce 2 parameters: `start` and `stop` times. With that the windows data will be created in python with something like this:
```python
import epics
from matplotlib import pyplot as plt

class RectWindow:
    def __init__(self, pv_name):
        # Create PV object
        self.pv = epics.PV(pv_name)

        # Get the waveform PV size
        self.size = self.pv.nelm

        # Flag to indicate a valid windows has been created
        self.valid = False

    def is_valid(self):
        return self.valid

    def set_size(self, start, stop):
        # Check bounds
        if (start >= stop) or (stop >= self.size):
            print('Invalid windows size!')
            self.valid = False
        else:
            self.valid_windows = True
            # Create rectangular window
            self.win = [0]*start + [1]*(stop-start) + [0]*(self.size-stop)
            # Preview the resulting window in the GUI
            plt.plot(self.win)
            plt.show()

    def write(self):
        if self.valid:
            # Write window data to PV
            self.pv.put(self.win)

if __name__ == "__main__":
    # This is the window waveform PV
    pv_name = 'test_jesus:win0'

    # These 2 parameters comes from the GUI
    start = 500
    stop = 3000

    rw = RectWindow(pv_name=pv_name)
    rw.set_size(start=start, stop=stop)

    # A 'write' button on the GUI will trigger this:
    r = raw_input("Do you want to write it? [y/n] ")
    if (r == "y"):
        rw.write()
```

The windows preview will show something like this:
![Screen Shot 2020-07-30 at 3 51 04 PM](https://user-images.githubusercontent.com/3729513/88982261-aa1c9900-d27c-11ea-860d-3e65c96724d8.png)

**Comment by jesusvasquez333** [comment]
These windows are now complex windows, so we 2 PVs: one for the real part, and one for the imaginary part.

**Comment by jesusvasquez333** [comment]
The PV for the complex windows are:
```
${DEVICE}:SIOC:B084:RF52:0:{I,Q}CPXWND{0..2}
```

Where:
- `I` : Real component window
- `Q`: Complex component windows
- `0..2` : are the windows indexes (we have 3 set of windows).

For example, for the RF52 station we have:
```bash
[jvasquez@rhel6-64b edm]$ caget SIOC:B084:RF52:0:{I,Q}CPXWND{0..2}.DESC
SIOC:B084:RF52:0:ICPXWND0.DESC I complex window
SIOC:B084:RF52:0:ICPXWND1.DESC I complex window
SIOC:B084:RF52:0:ICPXWND2.DESC I complex window
SIOC:B084:RF52:0:QCPXWND0.DESC Q complex window
SIOC:B084:RF52:0:QCPXWND1.DESC Q complex window
SIOC:B084:RF52:0:QCPXWND2.DESC Q complex window
```

**Comment by jesusvasquez333** [comment]
We might be able to use spicy to create the windows: https://docs.scipy.org/doc/scipy/reference/signal.windows.html

**Comment by jesusvasquez333** [comment]
This was implemented in #78. For now we implemented the rectangular windows which is the one people is currently using. We can implement other types of windows in the future if needed.
